src/propagation/serialization.jl

```diff
@@ -0,0 +1,204 @@
+@doc raw"""
+    TaylorInterpolantNSerialization{T}
+
+Custom serialization struct to save a `TaylorInterpolant{T, TaylorN{T}, 2}` to a `.jld2` file. 
+
+# Fields
+- `vars::Vector{String}`: jet transport variables. 
+- `order::Int`: order of Taylor polynomials.
+- `varorder::Int`: order of jet transport perturbations. 
+- `dims::Tuple{Int, Int}`: matrix dimensions. 
+- `t0::T`: initial time.
+- `t::Vector{T}`: vector of times. 
+- `x::Vector{T}`: vector of coefficients. 
+"""
+struct TaylorInterpolantNSerialization{T}
+    vars::Vector{String}
+    order::Int
+    varorder::Int
+    dims::Tuple{Int, Int}
+    t0::T
+    t::Vector{T}
+    x::Vector{T}
+end 
+
+# Tell JLD2 to save TaylorInterpolant{T, TaylorN{T}, 2} as TaylorInterpolantNSerialization{T}
+writeas(::Type{TaylorInterpolant{T, TaylorN{T}, 2}}) where {T} = TaylorInterpolantNSerialization{T}
+
+# Convert method to write .jld2 files
+function convert(::Type{TaylorInterpolantNSerialization{T}}, eph::TaylorInterpolant{T, TaylorN{T}, 2}) where {T}
+    # Variables 
+    vars = TS.get_variable_names()
+    # Number of variables 
+    n = length(vars)
+    # Matrix dimensions
+    dims = size(eph.x)
+    # Number of elements in matrix
+    N = dims[1] * dims[2]
```

Comment:
Would this be equivalent to `N = length(eph.h)`? Most of the time I'd imagine both work out fine, but `length`, if it works, is maybe a bit more generic.

Reply:
I've changed it to `N = length(eph.x)`.